rfcs/2022-04-09-accessibility.md

```diff
@@ -0,0 +1,1001 @@
+- For legibility, always use the same width numbers. As we deal in the financial ecosystem, it makes it more readable when we read numbers in combination. We should always use a monospaced number font. It should not necessarily be a coding font, it can be a generic font as well.
+- Don’t use underlines in any state if they are not links.
+- Limit the font-style variations. Try to keep things consistent. Do not overuse Bolds, All caps, etc.
+- Use markup wherever required. Do not use it for anything except code snippets.
+- Don’t make the content (block of text) width too long. Wrap,
+  - Titles: at max. of 70% of the page width.
+  - Subtitles & Paras: at max. of 50%-60% of the page width.
+- Make sure the text doesn’t overlap in between two different sections.
+
+> For more information on design accessibility you can check out component specific [design boards in figma](https://www.figma.com/file/LV9kjgzhkfsktvbRgTuGJ3/Blade---Base-DSL?node-id=557%3A14)
+
+**Tools:**
+
+- Material Design [Color Tool](https://material.io/resources/color/)
+
+# Tools & Infrastructure
+
+To ensure that blade components & apps built with blade are accessible it is essential that we make use of modern tooling and automated tests for accessibility testing. 
+
+Here are few of the must have tools that we will be implementing: 
+
+**Automated Testing:**
+
+- Automated unit [testing with jest-axe](https://github.com/nickcolley/jest-axe)
+- Static analysis with [eslint-plugin-jsx-a11y](https://github.com/jsx-eslint/eslint-plugin-jsx-a11y)
+- Storybook addon a11y for component level [UI testing](https://storybook.js.org/addons/@storybook/addon-a11y)
+- Using getBy `role`, `labelText`, `altText`, `title` while testing which encourages [proper accessibility testing](https://testing-library.com/docs/dom-testing-library/api-accessibility/)
+- Using [WAVE](https://wave.webaim.org/) for manual full page checks & finding accessibility issues.
+- Using [Accessibility Insights for Web Chrome extension](https://chrome.google.com/webstore/detail/accessibility-insights-fo/pbjjkligggfmakdaogkfomddhfmpjeni) *This one is very good*
+
+**Manual Testing:**
+
+Automated accessibility tools only [pick up around 40% of errors automatically](https://alphagov.github.io/accessibility-tool-audit/) at best, to ensure blade components are accessible it's essential to do manual testing
+
+- Follow preliminary guide for ["A First Review of Web Accessibility"](https://www.w3.org/WAI/test-evaluate/preliminary/) by w3c
+- Follow [WAI-ARIA Authoring Practices Guidelines](https://www.w3.org/WAI/ARIA/apg/patterns/) for component specific accessibility testing
+- Do [manual keyboard testing](#manual-testing)
+- Do manual [screen reader testing](https://dequeuniversity.com/screenreaders/voiceover-ios-shortcuts)
+- Follow a checklist for general page wide accessibility testing
```

Comment:
can we create a simple checklist optimized for our use cases and put it in some central place? every time we build a new component it's difficult for devs to go through so many things. Basically a checklist of items that all the devs shall use when building a component.

Reply:
Yes that makes sense, I was in fact planning to write a doc for that. But not entirely sure if Accessibility RFC is the right place to do this. Might be better if we create something like `component-development-lifecycle.md` file to document our process.

Reply:
Decided to decouple this as a different task to not block a11y RFC.